src/ESM.sol

```diff
@@ -55,23 +62,24 @@ contract ESM {
         require(z >= x);
     }
 
-    function fire() external {
-        require(fired == 0,  "esm/already-fired");
-        require(Sum >= min,  "esm/min-not-reached");
+    function fire(bool denyProxy) external {
```

Comment:
I worry this creates a race condition--unless the group triggering ES coordinates in another contract to reach critical mass and then `join`+`fire` atomically, or a single whale takes that action atomically, a malicious actor can try to front-run  with a call to `fire(false)` and retain governance control (they might even race to be the one putting in the final MKR). The simplest and safest option to have no option and always revoke the proxy.

Other options:
- build a coordination mechanism into this contract, e.g. have `join` accept a boolean for `denyProxy` too; MKR for the "deny" and "don't deny cases is then tallied separately
- build a separate coordination contract so MKR holders will not have to deploy one hastily in an emergency...I like the option above (just building it in here) better
- have two completely separate ESMs, one that denies the proxy and one that doesn't

In any case, I don't think we should count on a single whale acting atomically or MKR holders scraping together a coordination contract to do atomic `join`+`fire` in an emergency.

Reply:
Yes, this will work only in an atomic transaction. I was assuming we could figure something out externally as you mention in the second point, as maybe is not the best idea to join MKR until you have the enough amount anyway.
But if you guys consider we should give a better alternative I'll explore the other two options. I think I'll prefer to go the path of two different `esm` to keep the code simpler.

Reply:
I mean two different `esm` created from the same code with some variable in the constructor.

Reply:
Changes done.